src/Microsoft.ML.Api/DataViewConstructionUtils.cs

```diff
@@ -174,12 +174,7 @@ private Delegate CreateGetter(int index)
                         else if (outputType.GetElementType() == typeof(bool))
                         {
                             Ch.Assert(colType.ItemType.IsBool);
-                            return CreateConvertingArrayGetterDelegate<bool, DvBool>(index, x => x);
-                        }
-                        else if (outputType.GetElementType() == typeof(bool?))
-                        {
-                            Ch.Assert(colType.ItemType.IsBool);
-                            return CreateConvertingArrayGetterDelegate<bool?, DvBool>(index, x => x ?? DvBool.NA);
+                            return CreateConvertingArrayGetterDelegate<bool, bool>(index, x => x);
```

Comment:
>bool [](start = 77, length = 4)

you don't have to do it.
just fall back to default T[] -> VBuffer<T> code.

Reply:
of course.

---
In reply to: [211109130](https://github.com/dotnet/machinelearning/pull/692#discussion_r211109130) [](ancestors = 211109130)